ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -1300,6 +1300,33 @@ private class JdbcLedgerDao(
     )
   }
 
+  override def storePackageRejection(
+      participantId: ParticipantId,
+      submissionId: String,
+      reason: String
+  ) = {
+    val prereqs = Try {
+      require(participantId.nonEmpty, "participantId cannot be empty")
+      require(submissionId.nonEmpty, "submissionId cannot be empty")
+    }
+    prereqs.fold(
+      Future.failed,
+      _ =>
+        dbDispatcher.executeSql("store package rejections") { implicit conn =>
+          val sqlupdate = executeBatchSql(
```

Comment:
this doesn't seem to be updating the external offset. If the indexer would die after processing this it would end up re-indexing this entry, which would likely fail because of a (participant_id, submission_id) uniqueness constraint, which would then end up making the indexer stuck (we can never proceed if we fail to index an entry as that might mean we're skipping over events, which would just end up failures down the line)

Reply:
See https://github.com/digital-asset/daml/blob/9f9d2d0458013cb8f75dfc5c630a1b711c38359a/ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala#L708